lib/src/interfaces/a_star.dart

```diff
@@ -54,13 +58,88 @@ class AutonomyAStarState extends AStarState<AutonomyAStarState> {
     orientation: orientation, 
     direction: direction,
     goal: goal, 
-    depth: direction == DriveDirection.forward ? depth + 1 : depth + 2,
+    depth: (direction == DriveDirection.forward)
+        ? depth + 1
+        : (direction == DriveDirection.forwardLeft || direction == DriveDirection.forwardRight)
+            ? depth + sqrt2
+            : depth + 2,
   );
 
+  bool drivingThroughObstacle(AutonomyAStarState state) {
+    final isTurn = state.direction != DriveDirection.forward;
+    final isQuarterTurn = state.direction == DriveDirection.forwardLeft || state.direction == DriveDirection.forwardRight;
+
+    // Forward drive across the perpendicular axis
+    if (!isTurn && state.orientation.angle.abs() % 90 == 0) {
+      return false;
+    }
+
+    // Not encountering any sort of diagonal angle
+    if (isTurn && isQuarterTurn && state.orientation.angle.abs() % 90 == 0) {
+      return false;
+    }
+
+    // No diagonal movement, won't drive between obstacles
+    if (!isQuarterTurn && orientation.angle.abs() % 90 == 0) {
+      return false;
+    }
+
+    DriveOrientation orientation1;
+    DriveOrientation orientation2;
```

Comment:
What do these two represent?

Reply:
Okay I think I get it, just elaborate a bit more in a full-on doc comment `///` or in-function comment `//`. Doc comment if the caller needs to know, regular comment otherwise. 

One trick I use for complicated logic is explicitly naming a few cases in the comments, then referencing them in the code, like this:

```dart
// There are a few cases this function handles: 
//
// 1. The first thing occurs
// 2. The second thing occurs
// 3. Maybe another thing

if (...) {  // case 1
  // ... 
} else if (...) {  // case 2
  // ...
} 

return null;  // case 3